workers/social/lib/social/models/account.coffee

```diff
@@ -585,13 +585,16 @@ module.exports = class JAccount extends jraphical.Module
     unless delegate.can 'verify-emails'
       callback new KodingError 'Access denied'
     else
-      JUser = require './user'
-      JUser.one { username }, (err, user) ->
-        return  callback err if err
-        return  callback new Error 'User is not found' unless user
-        user.confirmEmail (err) ->
-          return callback new Error 'An error occurred while confirming email' if err
-          callback null, yes
+      @findUserAndConfirmEmail username, callback
+
+  findUserAndConfirmEmail: (username, callback) ->
+    JUser = require './user'
+    JUser.one { username }, (err, user) ->
+      return  callback err if err
+      return  callback new Error 'User is not found' unless user
+      user.confirmEmail (err) ->
+        return callback new Error 'An error occurred while confirming email' if err
+        callback null, yes
```

Comment:
let's revert this change as well after: https://github.com/koding/koding/pull/8977/files#r78067181